Detectors/TPC/calibration/run/calib-pedestal.cxx

```diff
@@ -0,0 +1,93 @@
+  WorkflowSpec workflow;
+  workflow.emplace_back(DataProcessorSpec{
+    "calib-pedestal",
+    select(config.options().get<std::string>("input-spec").c_str()),
+    Outputs{},
+    AlgorithmSpec{[](InitContext& setup) { return adaptStateless([](InputRecord& inputs, DataAllocator& outputs) {
+                                             for (auto& input : inputs) {
+                                               const auto* dh = DataRefUtils::getHeader<o2::header::DataHeader*>(input);
+                                               LOG(INFO) << dh->dataOrigin.as<std::string>() << "/" << dh->dataDescription.as<std::string>() << "/"
+                                                         << dh->subSpecification << " payload size " << dh->payloadSize;
+
+                                               // there is a bug in InpuRecord::get for vectors of simple types, not catched in
+                                               // DataAllocator unit test
+                                               //auto data = inputs.get<std::vector<char>>(input.spec->binding.c_str());
+                                               //LOG(INFO) << "data size " << data.size();
+                                               printHeader();
+                                               try {
+                                                 const char* pos = input.payload;
+                                                 const char* last = pos + dh->payloadSize;
+
+                                                 //o2::framework::RawParser parser(input.payload, dh->payloadSize);
+
+                                                 while (pos < last) {
+                                                   //for (auto it = parser.begin(), end = parser.end(); it != end; ++it) {
+                                                   const auto& rdh = *((RDH*)pos);
+                                                   const auto payloadSize = rdh.memorySize - rdh.headerSize;
+                                                   const auto dataWrapperID = rdh.endPointID;
+                                                   const auto linkID = rdh.linkID;
+                                                   const auto globalLinkID = linkID + dataWrapperID * 12;
+                                                   //LOG(INFO) << fmt::format("CRU: {:3d}, link: {:2d}, size: {:5d}", rdh.cruID, globalLinkID, payloadSize);
+                                                   printRDH(rdh);
+
+                                                   pos += rdh.offsetToNext;
+                                                 }
+                                               } catch (const std::runtime_error& e) {
+                                                 LOG(ERROR) << "can not create raw parser form input data";
+                                                 o2::header::hexDump("payload", input.payload, dh->payloadSize, 64);
+                                                 throw e;
```

Comment:
```suggestion
                                                 LOG(ERROR) << e.what();
```
Note that I have replaced this in the meanwhile in the raw-parser example. This will allow you to keep on stepping through the input data if the parser could not be created for one block.

Reply:
done, however this is just the initial step. I'll still need to implement the actual calibration.